Clarify Server/Message Management Description

chevyboys opened the pull request:
The current API documentation of the manage server permission is somewhat vague, and doesn't actually describe the specific actions someone could take when granted that role. In addition, manage messages fails to mention pinning messages in it's current form, which could be confusing to beginning bot developers 

This change fixes both issues, making things a little bit more clear on both permissions

chevyboys committed:
Clarify Server/Message Management Description

The current API documentation of the manage server permission is somewhat vague, and doesn't actually describe the specific actions someone could take when granted that role. In addition, manage messages fails to mention pinning messages in it's current form, which could be confusing to beginning bot developers 

This change fixes both issues, making things a little bit more clear on both permissions

chevyboys committed:
Fix small spelling mistake

vvito7 reviewed (commented):
You should use "guild", not "server".

advaith1 commented on docs/topics/Permissions.md:
```diff
@@ -33,15 +33,15 @@ Below is a table of all current permissions, their integer values in hexadecimal
 | ADMINISTRATOR \*              | `0x0000000000000008` `(1 << 3)`  | Allows all permissions and bypasses channel permission overwrites                                                                                   |              |
 | MANAGE_CHANNELS \*            | `0x0000000000000010` `(1 << 4)`  | Allows management and editing of channels                                                                                                           | T, V, S      |
-| MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows management and editing of the guild                                                                                                          |              |
+| MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows management of the server name, region, and adding bots to the server                                                                          |              |
```
server region doesn't exist anymore, and manage server does a lot more than what is listed here.

chevyboys replied:
Fair. I'm updating that in my latest commit involving most of the suggestions made here

chevyboys replied:
I would rather err on the side of being more specific then more vague, let me know if there is anything still missing from manage server in the latest commit.

advaith1 commented on docs/topics/Permissions.md:
```diff
@@ -33,15 +33,15 @@ Below is a table of all current permissions, their integer values in hexadecimal
 | SEND_MESSAGES                 | `0x0000000000000800` `(1 << 11)` | Allows for sending messages in a channel (does not allow sending messages in threads)                                                               | T            |
 | SEND_TTS_MESSAGES             | `0x0000000000001000` `(1 << 12)` | Allows for sending of `/tts` messages                                                                                                               | T            |
-| MANAGE_MESSAGES \*            | `0x0000000000002000` `(1 << 13)` | Allows for deletion of other users messages                                                                                                         | T            |
+| MANAGE_MESSAGES \*            | `0x0000000000002000` `(1 << 13)` | Allows for deletion of or pinning other users messages                                                                                             | T            |
```
```suggestion
| MANAGE_MESSAGES \*            | `0x0000000000002000` `(1 << 13)` | Allows for deletion of other users' messages and pinning messages                                                                                   | T            |
```

AlmostSuspense replied:
it also allows bypassing slowmode and publishing all messages in news channels

almeidx replied:
and removing reactions

advaith1 replied:
yeah... maybe this should just say "managing messages"

chevyboys replied:
> yeah... maybe this should just say "managing messages"

I would rather be more specific then that, since someone may be looking in these documents for specific permissions to request for a specific task (such as deleting reactions) and I would like that to be supported as fully as is reasonably possible in the documentation.

PlavorSeol commented on docs/topics/Permissions.md:
```diff
@@ -33,15 +33,15 @@ Below is a table of all current permissions, their integer values in hexadecimal
 | ADMINISTRATOR \*              | `0x0000000000000008` `(1 << 3)`  | Allows all permissions and bypasses channel permission overwrites                                                                                   |              |
 | MANAGE_CHANNELS \*            | `0x0000000000000010` `(1 << 4)`  | Allows management and editing of channels                                                                                                           | T, V, S      |
-| MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows management and editing of the guild                                                                                                          |              |
+| MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows management of the server name, region, and adding bots to the server                                                                          |              |
```
```suggestion
| MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows deleting invites, editing guild settings and welcome screen and managing guild template, integrations and widget                             |              |
```

Jupith replied:
"editing guild settings" *includes* welc screen, template, widget and integrations, i think its just easier to leave it as Allows management and editing of the guild

PlavorSeol replied:
Editing [guild settings](https://discord.com/developers/docs/resources/guild#modify-guild) and [welcome screen](https://discord.com/developers/docs/resources/guild#modify-guild-welcome-screen) and managing [guild template](https://discord.com/developers/docs/resources/guild-template#modify-guild-template), [integrations](https://discord.com/developers/docs/resources/guild#delete-guild-integration) and [widget](https://discord.com/developers/docs/resources/guild#modify-guild-widget) are all different things in the term of API, they use clearly *different* endpoints, and my suggestion reflects them, assuming OP's intention is to clarify things.

chevyboys replied:
> Editing [guild settings](https://discord.com/developers/docs/resources/guild#modify-guild) and [welcome screen](https://discord.com/developers/docs/resources/guild#modify-guild-welcome-screen) and managing [guild template](https://discord.com/developers/docs/resources/guild-template#modify-guild-template), [integrations](https://discord.com/developers/docs/resources/guild#delete-guild-integration) and [widget](https://discord.com/developers/docs/resources/guild#modify-guild-widget) are all different things in the term of API, they use clearly _different_ endpoints, and my suggestion reflects them, assuming OP's intention is to clarify things.

Is there a way to insert those links into the docs so people can see those descriptions of what each of those things entails?

PlavorSeol commented on docs/topics/Permissions.md:
```diff
@@ -33,15 +33,15 @@ Below is a table of all current permissions, their integer values in hexadecimal
 | SEND_MESSAGES                 | `0x0000000000000800` `(1 << 11)` | Allows for sending messages in a channel (does not allow sending messages in threads)                                                               | T            |
 | SEND_TTS_MESSAGES             | `0x0000000000001000` `(1 << 12)` | Allows for sending of `/tts` messages                                                                                                               | T            |
-| MANAGE_MESSAGES \*            | `0x0000000000002000` `(1 << 13)` | Allows for deletion of other users messages                                                                                                         | T            |
+| MANAGE_MESSAGES \*            | `0x0000000000002000` `(1 << 13)` | Allows for deletion of or pinning other users messages                                                                                             | T            |
```
```suggestion
| MANAGE_MESSAGES \*            | `0x0000000000002000` `(1 << 13)` | Allows crossposting other users' messages in news channels, deleting other users' messages and reactions and pinning messages                       | T            |
```

chevyboys replied:
I'm manually adding this since it conflicted with another suggestion.

chevyboys committed:
Update docs/topics/Permissions.md

Co-authored-by: Advaith <advaithj1@gmail.com>

chevyboys committed:
Update docs/topics/Permissions.md

Editing guild settings and welcome screen and managing guild template, integrations and widget are all different things in the term of API, they use clearly different endpoints, and my suggestion reflects them, assuming OP's intention is to clarify things.

Co-authored-by: PlavorSeol <pseol2190@outlook.com>

chevyboys committed:
Integrate suggestions

use "guild" instead of "server", and expand the descriptions further to describe more fully what each permission entails

Jupith commented on docs/topics/Permissions.md:
```diff
@@ -33,20 +33,20 @@ Below is a table of all current permissions, their integer values in hexadecimal
 | ADMINISTRATOR \*              | `0x0000000000000008` `(1 << 3)`  | Allows all permissions and bypasses channel permission overwrites                                                                                   |              |
 | MANAGE_CHANNELS \*            | `0x0000000000000010` `(1 << 4)`  | Allows management and editing of channels                                                                                                           | T, V, S      |
-| MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows management and editing of the guild                                                                                                          |              |
+| MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows deleting invites, editing guild settings, guild welcome screen, and managing a guild's template, integrations (bots, webhooks, and app integrations), and the guild widget                             |              |
```
```suggestion
| MANAGE_GUILD \*               | `0x0000000000000020` `(1 << 5)`  | Allows management and editing of the guild                                                               |              |
```

chevyboys replied:
@Jupith, since editing an management of the guild isn't terribly helpful for someone who doesn't already know what that means, I feel that something more descriptive would be more beneficial. Unless there is an issue with formatting, is there a good reason I'm missing for shortening the description?

AlmostSuspense replied:
There could be a section describing what each extended permission does (by that i mean the ones that give access to multiple things, like Manage Server/Messages) so the table doesn't become too big, similar to [this](https://discord.com/developers/docs/topics/permissions#implicit-permissions)

chevyboys replied:
@Suspense4615  What if it were changed to `Allows managing invites, settings, welcome screen, template, integrations (bots, webhooks, and app integrations), and the widget of a guild`? Then its about the same length as the description in the moderate members permission. It seems like getting up to three lines isn't a huge deal, and the added clarity would be a big improvement, without having to create a new section.

chevyboys replied:
Alternatively, if we used @PlavorSeol 's suggestion from above, (quoted below) and had each of his categories link to the relevant endpoint information, then that would shorten things and ensure detailed descriptions

He suggested we change it to `Allows deleting invites, editing guild settings and welcome screen and managing guild template, integrations and widget`

His comment on that was:
> Editing [guild settings](https://discord.com/developers/docs/resources/guild#modify-guild) and [welcome screen](https://discord.com/developers/docs/resources/guild#modify-guild-welcome-screen) and managing [guild template](https://discord.com/developers/docs/resources/guild-template#modify-guild-template), [integrations](https://discord.com/developers/docs/resources/guild#delete-guild-integration) and [widget](https://discord.com/developers/docs/resources/guild#modify-guild-widget) are all different things in the term of API, they use clearly _different_ endpoints, and my suggestion reflects them, assuming OP's intention is to clarify things.

AlmostSuspense replied:
This is not taking upcoming features into account, while it fits with 3 lines *now*, Discord might add several features tied to that permission in the future. And I don't think adding a few lines once every now and then is sustainable on the long run

chevyboys replied:
@Suspense4615 Can you show me an example of what you think would work best?

AlmostSuspense replied:
> @Suspense4615 Can you show me an example of what you think would work best?

something like [this](https://discord.com/developers/docs/topics/permissions#implicit-permissions)

chevyboys committed:
Add bypassing slowmode to manage messages
